doc/decisions/DR020-Coding-style.md

```diff
@@ -0,0 +1,68 @@
+# DR020 Coding style
+
+- **Status:** Decided
+- **Driver:** @foundrytom
+- **Approver:** @felltech @carmenpinto @fnRaihanKibria
+- **Outcome:** OpenAssetIO will use the Google C++ Style Guide.
+
+## Background
+
+OpenAssetIO is a collaboratively maintained codebase. A consistent
+coding style can help simplify maintenance. As such, contributors need
+clear guidelines.
+
+## Options considered
+
+### Option 1 - Define a custom style guide
+
+Maintainers define a bespoke style guide that embodies their
+locally agreed preferences.
+
+#### Pros
+
+- Allows local consensus around what is preferable in the code base.
+
+#### Cons
+
+- Consensus is nigh on impossible when it comes to subjective choices
+  such as code style, as the number of contributors grows beyond one.
+  Often resulting in unbalanced compromise.
+- Disagreements over the style may well be directly attributable to a
+  specific individual within the community, that then becomes the target
+  of criticism.
+- Local control over style minutiae adds the temptation to futz with it
+  over time, ultimately reducing overall consistency.
+- May increase the complexity of linter configuration when attempting
+  to encode the agreed standard.
+
+### Option 2 - Use a common well-established standard
+
+Pick a well known existing standard and just use that.
+
+#### Pros
```

Comment:
[consider] 

* Can always be tweaked to better suit the team (e.g. line length limits) or to cover additional areas that the standard has no opinion on.

Reply:
Added [https://github.com/OpenAssetIO/OpenAssetIO/pull/1096/commits/b4e33b673288fc113cbf740060a6cb9e739e1e2b](here).